src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -296,6 +323,77 @@ private ODataResourceSet ReadFeedCore(bool lazy)
             return result;
         }
 
+        /// <summary>
+        /// Reads an <see cref="ODataDeltaResourceSet"/>.
+        /// </summary>
+        /// <returns>The <see cref="MaterializerDeltaFeed"/> of the read <see cref="ODataDeltaResourceSet"/>.</returns>
+        private MaterializerDeltaFeed ReadDeltaFeedCore()
+        {
+            this.ExpectState(ODataReaderState.DeltaResourceSetStart);
+
+            ODataDeltaResourceSet result = (ODataDeltaResourceSet)this.reader.Item;
+            MaterializerDeltaFeed feed = null;
+
+            Debug.Assert(this.items != null, "this.items should not be null");
+
+            if (result == null)

```

Comment:
Curious how or when this situation can arise...

Reply:
remove it.